packages/common/src/lib/notify/notify-builder.js

```diff
@@ -1,184 +1,214 @@
 const logger = require('../logger');
 const { createNotifyClient } = require('./notify-factory');
 
-function NotifyBuilder() {
-  function setPrivateNotifyClient(notifyClient) {
+module.exports = {
+  /**
+   * Resets all the data on the object useful for cleaning the client between emails or tests.
+   *
+   * @returns {*}
+   */
+  reset() {
+    this.notifyClient = undefined;
+    this.templateId = undefined;
+    this.templatePersonalisation = undefined;
+    this.destinationEmail = undefined;
+    this.emailReplyToId = undefined;
+    this.reference = undefined;
+
+    return this;
+  },
+
+  setPrivateNotifyClient(notifyClient) {
     logger.debug({ notifyClient }, 'Setting notify client');
     this.notifyClient = notifyClient;
-  }
+  },
 
   /**
-   * Intentionally do not expose the notify client directly.
+   * Set the default notify client. This should always be the first function called unless the intent is to mock the client.
```

Comment:
Comment is misleading ?

This doesn't seem to be the case for all calls in this PR - maybe I have misunderstood.

Reply:
I dont follow what do you propose it should be?